apps/prairielearn/src/components/QuestionsTable.html.ts

```diff
@@ -0,0 +1,207 @@
+import { html, HtmlSafeString } from '@prairielearn/html';
+import { nodeModulesAssetPath, compiledScriptTag } from '../lib/assets';
+import { EncodedData } from '@prairielearn/browser-utils';
+
+export function QuestionsTable(
+  questions,
+  showAddQuestionButton,
+  current_course_instance,
+  course_instances,
+  urlPrefix,
+  plainUrlPrefix,
+  errorMessage,
+): HtmlSafeString {
+  const has_legacy_questions = questions.some((row) => row.display_type !== 'v3');
+  const course_instance_ids = (course_instances || []).map((course_instance) => course_instance.id);
+  return html`
+    ${EncodedData(course_instance_ids, 'course-instance-ids')};
+    ${EncodedData(showAddQuestionButton, 'show-add-question-button')}
+    ${EncodedData(urlPrefix, 'url-prefix')} ${EncodedData(plainUrlPrefix, 'plain-url-prefix')}
+    <style>
+      .sticky-column {
+        position: sticky;
+        left: 0;
+        background: white;
+        background-clip: padding-box;
+        box-shadow: inset -1px 0 #dee2e6;
+      }
+      .table-hover tbody tr:hover td.sticky-column {
+        color: #212529;
+        background-color: #efefef;
+      }
+      .fixed-table-toolbar {
+        padding: 0 1em 0 1em;
+      }
+      .fixed-table-toolbar div.pagination,
+      .fixed-table-toolbar div.pagination-detail {
+        margin: 0 1em 0 0 !important;
+      }
+    </style>
+
+    <script src="${nodeModulesAssetPath('lodash/lodash.min.js')}"></script>
+    <script src="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.js')}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.js',
+      )}"></script>
+    <script src="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.js',
+      )}"></script>
+    <link
+      href="${nodeModulesAssetPath('bootstrap-table/dist/bootstrap-table.min.css')}"
+      rel="stylesheet"
+    />
+    <link
+      href="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/sticky-header/bootstrap-table-sticky-header.min.css',
+      )}"
+      rel="stylesheet"
+    />
+    <link
+      href="${nodeModulesAssetPath(
+        'bootstrap-table/dist/extensions/filter-control/bootstrap-table-filter-control.min.css',
+      )}"
+      rel="stylesheet"
+    />
+
+    ${compiledScriptTag('questionsTable.js')}
+
+    <main id="content" class="container-fluid">
```

Comment:
I'd typically recommend keeping `<main>` out of components; it should be up to the parent to decide what `<main>` is, how wide the container should be, etc.

Reply:
done